File: src/Api/Controllers/CommandsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace Api.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class CommandsController : ControllerBase
    {
        private CommandHandler.EventStore eventStore;

        public CommandsController(CommandHandler.EventStore eventStore)
        {
            this.eventStore = eventStore;
        }

        [HttpPost("addbookie")]
        public IActionResult AddBookie([FromBody] Dtos.AddBookieDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("makedeposit")]
        public IActionResult MakeDeposit([FromBody] Dtos.MakeDepositDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("makewithdrawal")]
        public IActionResult MakeWithdrawal([FromBody] Dtos.MakeWithdrawalDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("placebackbet")]
        public IActionResult PlaceBackBet([FromBody] Dtos.PlaceBackBetDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("placelaybet")]
        public IActionResult PlaceLayBet([FromBody] Dtos.PlaceLayBetDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("placefreebet")]
        public IActionResult PlaceFreeBet([FromBody] Dtos.PlaceFreeBetDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("settlebackBet")]
        public IActionResult SettleBackBet([FromBody] Dtos.SettleBackBetDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("settlefreeBet")]
        public IActionResult SettleFreeBet([FromBody] Dtos.SettleFreeBetDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("settlelayBet")]
        public IActionResult SettleLayBet([FromBody] Dtos.SettleLayBetDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("cashoutbackbet")]
        public IActionResult CashOutBackBet([FromBody] Dtos.CashOutBackBetDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }

        [HttpPost("creditbonus")]
        public IActionResult CreditBonus([FromBody] Dtos.CreditBonusDto commandDto)
        {
            return ExecuteCommand(commandDto);
        }
 
        private IActionResult ExecuteCommand(Dtos.CommandDto domainCommand)
        {
            var executionResult = CommandHandler.handle(eventStore, domainCommand);
            if(executionResult.IsOk) {
                var state = CommandHandler.getCurrentState(eventStore, domainCommand.AggregateId);
                Console.WriteLine(state);
                return Ok();
            } else {
                Console.WriteLine(executionResult.ErrorValue);
                return Conflict(executionResult.ErrorValue);
            }
        }
    }
}

File: src/Domain.Api/Controllers/CommandsController.fs
namespace Domain.Api.Controllers

open Microsoft.AspNetCore.Mvc

[<Route("api/[controller]")>]
[<ApiController>]
type CommandsController (eventStore: CommandHandler.EventStore) =
    inherit ControllerBase()

    let eventStore = eventStore

    member private this.ExecuteCommand(domainCommand: CommandDtos.CommandDto): IActionResult =
        let executionResult = CommandHandler.handle eventStore domainCommand
        match executionResult with
        | Ok _ ->
            let state = CommandHandler.getCurrentState eventStore domainCommand.AggregateId
            printfn "%A" state
            this.Ok() :> IActionResult
        | Error x ->
            printfn "%A" domainCommand.AggregateId
            printfn "%A" x
            this.Conflict(x) :> IActionResult

    [<HttpPost("addbookie")>]
    member this.AddBookie([<FromBody>] commandDto:CommandDtos.AddBookieDto) =
        this.ExecuteCommand(commandDto)

    [<HttpPost("makedeposit")>]
    member this.MakeDeposit([<FromBody>] commandDto:CommandDtos.MakeDepositDto) =
        this.ExecuteCommand commandDto
    
    [<HttpPost("makewithdrawal")>]
    member this.MakeWithdrawal([<FromBody>] commandDto:CommandDtos.MakeWithdrawalDto) =
        this.ExecuteCommand commandDto
    
    [<HttpPost("placebackbet")>]
    member this.PlaceBackBet([<FromBody>] commandDto:CommandDtos.PlaceBackBetDto) =
        this.ExecuteCommand commandDto
    
    [<HttpPost("placelaybet")>]
    member this.PlaceLayBet([<FromBody>] commandDto:CommandDtos.PlaceLayBetDto) =
        this.ExecuteCommand commandDto
    
    [<HttpPost("placefreebet")>]
    member this.PlaceFreeBet([<FromBody>] commandDto:CommandDtos.PlaceFreeBetDto) =
        this.ExecuteCommand commandDto
    
    [<HttpPost("settlebackBet")>]
    member this.SettleBackBet([<FromBody>] commandDto:CommandDtos.SettleBackBetDto) =
        this.ExecuteCommand commandDto

    [<HttpPost("settlefreeBet")>]
    member this.SettleFreeBet([<FromBody>] commandDto:CommandDtos.SettleFreeBetDto) =
        this.ExecuteCommand commandDto

    [<HttpPost("settlelayBet")>]
    member this.SettleBackBet([<FromBody>] commandDto:CommandDtos.SettleLayBetDto) =
        this.ExecuteCommand commandDto

    [<HttpPost("cashoutbackbet")>]
    member this.CashOutBackBet([<FromBody>] commandDto:CommandDtos.CashOutBackBetDto) =
        this.ExecuteCommand commandDto

    [<HttpPost("creditbonus")>]
    member this.CreditBonus([<FromBody>] commandDto:CommandDtos.CreditBonusDto) =
        this.ExecuteCommand commandDto

    [<HttpGet("{id}")>]
    member this.Get(id:int) =
        let value = "value"
        ActionResult<string>(value)

    
